### PR TITLE
make python code compatible with both python2 and python3

### PR DESCRIPTION
**Pull request by nlu90**
Given the deadline for python3 migration is drawing near, we need to move all python related code to python3 now.

This PR uses the 2to3 tools to update all the python files.

### CODE REVIEW

**Comment by nwangtw** [comment]
There are a few syntax errors to fix

**Comment by nlu90** [comment]
> There are a few syntax errors to fix

It seems some syntax error only reported undert python2's syntax checker.

```# nlu @ tw-mbp-nlu in ~/workspace/heron on git:neng/py3-migration o [9:39:33]
$ find . -name "*.py" | xargs python -m py_compile
  File "./heron/tools/ui/src/python/main.py", line 190
    print('\n', end='')
                   ^
SyntaxError: invalid syntax

  File "./heron/tools/tracker/src/python/main.py", line 274
    print('\n', end='')
                   ^
SyntaxError: invalid syntax

  File "./heron/tools/cli/src/python/result.py", line 81
    print(msg, file=f)
                   ^
SyntaxError: invalid syntax

  File "./heron/statemgrs/src/python/statemanager.py", line 36
    class StateManager(metaclass=abc.ABCMeta):
                                ^
SyntaxError: invalid syntax
```

But with python3's syntax checker, those errors are gone.
```
# nlu @ tw-mbp-nlu in ~/workspace/heron on git:neng/py3-migration o [9:39:37] C:1
$ find . -name "*.py" | xargs python3 -m py_compile

# nlu @ tw-mbp-nlu in ~/workspace/heron on git:neng/py3-migration o [9:40:21]
$
```

**Comment by nwangtw** [comment]
Seems like cpplint is not compatible? Maybe try to upgrade it https://github.com/apache/incubator-heron/tree/master/third_party/python/cpplint

**Comment by nlu90** [comment]
> Seems like cpplint is not compatible? Maybe try to upgrade it https://github.com/apache/incubator-heron/tree/master/third_party/python/cpplint

The original `cpplint.py` file is actually compatible with both 2 and 3 with a built-in rename hack. So most of the changes to it should be reverted.

File: heron/shell/src/python/handlers/killexecutorhandler.py [hunk 1]
```diff
@@ -20,10 +20,13 @@
...
 
 ''' killexecutorhandler.py '''
+from future.standard_library import install_aliases
+install_aliases()
+
 import logging
 import os
 import signal
-import urlparse
+from urllib.parse import parse_qsl
```
Comment (xiaoyao1991) [inline heron/shell/src/python/handlers/killexecutorhandler.py]: Looks like this is a package only valid in py3?

Comment (nlu90) [reply heron/shell/src/python/handlers/killexecutorhandler.py]: it's available in python2 with the first two lines import of this file

```
from future.standard_library import install_aliases
install_aliases()
```

**Pull request merged**